File: 0134-gas-station/0134-gas-station.py
class Solution:
  def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
    if sum(gas)-sum(cost) < 0: return -1
    
    total = ans = 0
    for i in range(len(gas)):
      total += gas[i]-cost[i]
      if total < 0:
        total = 0
        ans = i+1
    return ans

File: 0134-gas-station/0134-gas-station.java
class Solution {
  public int canCompleteCircuit(int[] gas, int[] cost) {
    int diff = IntStream.of(gas).sum() - IntStream.of(cost).sum();
    if (diff < 0) return -1;
    
    int total = 0;
    int ans = 0;
    for (int i = 0; i < gas.length; i++) {
      total += gas[i] - cost[i];
      if (total < 0) {
        total = 0;
        ans = i + 1;
      }
    }
    return ans;
  }
}